Update to Terms of Use description

David-Chadwick opened the pull request:
Give examples of ToU

This will address issue #1010 by being merged.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/David-Chadwick/vc-data-model/pull/1295.html" title="Last updated on Nov 4, 2023, 5:19 PM UTC (71f3de9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1295/77f95bc...David-Chadwick:71f3de9.html" title="Last updated on Nov 4, 2023, 5:19 PM UTC (71f3de9)">Diff</a>

decentralgabe commented:
this is at odds with #1294

TallTed commented on index.html:
```diff
@@ -3186,12 +3186,20 @@ <h3>Terms of Use</h3>
+Examples for the value of the <code>termsOfUse</code> <a>property</a> may be
+to tells the <a>verifier</a> 
+<li>
+- what actions it is required to perform (an <em>obligation</em>),
```
```suggestion
what actions it is required to perform (an <em>obligation</em>),
```

David-Chadwick replied:
@TallTed I have made the editorial changes you have requested, and the revised example has been taken from the EBSI documentation and has a real URL that can be resolved.

TallTed commented on index.html:
```diff
@@ -3186,12 +3186,20 @@ <h3>Terms of Use</h3>
+<a>verifiable presentation</a>;
+</li>
+<li>
+- what procedures or policies were used in issuing the verifiable credential, for example, by providing a pointer to the location where these procedures or policies can be found, or naming the standard that defines them.
```
```suggestion
what procedures or policies were used in issuing the verifiable credential,
for example, by providing a pointer to the location where these procedures
or policies can be found, or naming the standard that defines them.
```

TallTed reviewed (changes requested):
Change requests are assuming this text gets merged.

I want to see ACTUAL example URIs (and what might be retrieved by dereferencing those URIs) and/or sample textual values, not hand-waves about what might be communicated through such values.

msporny reviewed (changes requested):
Concrete ask: Please update the example to include a concrete implementation of ToU.

It's been stated here https://github.com/w3c/vc-data-model/issues/1010 that multiple implementations exist. Let's use a real example in the spec, wrap up the documentation for this section to speak directly to that real world example, and take this property off of the "at risk" list given the number of people claiming to have implemented it (with links to those implementations).

David-Chadwick commented:
@msporny There are 11 EBSI conformant wallets that are listed here
https://ec.europa.eu/digital-building-blocks/wikis/display/EBSI/Conformant+wallets
You can follow the links on the above page to all these implementations

davidlehn reviewed (commented):
Proof reading issues.

davidlehn commented:
(This got a bit long, perhaps needs to be moved to other issues.)
Some comments in general and related to the updated example:
- The "Terms of Use" section seems wholly incomplete.  How to author, use, or interpret `termsOfUse` isn't specified with any rigor.
- It kind of looks like `termsOfUse` is an escape hatch to shift processing rules to what is in the `type` field?  There are no details on how this works.
- Can references be added to the specs for types like `IssuanceCertificate` or `HolderPolicy`?  Or are those ad-hoc examples?  I think they resolve off the example `@vocab` now.  Should they resolve to real URLs?
- It's unclear to me what the `IssuanceCertificate` example has to do with "terms of use".  It seems like a method to verify an issuer?
- The `IssuanceCertificate` example text has some low level type-specific details.  Those would seem to be better in a spec that is referred to.  And the text there can be more general.  That this is some blockchain JWT JSON blob thing, with no spec references, is a bit much.
- The URLs in the examples look ephemeral.  Are they going to be around for a long time so future spec readers can use them?  (This is perhaps in conflict with the request for real links.)
- The name and text implies some sort of usage binding, and I suspect people will think it's legal binding.  Without any processing rules or type-specific rules specified, it's hard to say what the `termsOfUse` might mean at all, if anything.  This sort of thing seems beyond the scope of this spec but saying nothing seems confusing to implementors.
- The text says this is a "rapidly evolving feature" and refers to the implementation guide which doesn't mention `termsOfUse` at all.
- The text that this feature is "expected to be used by government-issued [VCs] ..." seems far too contemporary.  Especially for a feature marked at risk.  I think spec text here should say what a feature is for and how to use it.  Then let potential usage play out as it will.
- With all the above said, I do like the concept of indicating the intended use of VC data.  I think "terms" and "policy" terminology may be too strong for something that may be ignored or unenforceable.  This all needs to be spec'ed out much better, and perhaps that is best done in another spec doc with minimal text here.

David-Chadwick commented on index.html:
```diff
@@ -3186,12 +3186,22 @@ <h3>Terms of Use</h3>
         <p>
-The value of the <code>termsOfUse</code> <a>property</a> tells the
-<a>verifier</a> what actions it is required to perform (an <em>obligation</em>),
+Examples for the value of the <code>termsOfUse</code> <a>property</a> may be
+to tells the <a>verifier</a> 
```
Thankyou. I will change tells to tell

David-Chadwick commented:
@davidlehn All the extension points (Evidence, ToU, Refresh Service etc.)  have a type and the type defines the precise semantics of the extension and how it should be processed. So it very difficult (impossible?) to resolve the conflict between a general specification of the extension and the request to have a concrete example with working URLs and a detailed specification of the type with type specific processing rules. In my opinion the sensible option is to leave the VCDM with the generic definition and then in the Directory of extensions to specify particular types in detail and how they can be processed.

Sakurann reviewed (changes requested):
adding an example does not make ToU "better defined" and does not solve #1010

OR13 commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
+  validFrom: "2021-11-01T00:00:00Z",
+  issued: "2021-10-30T00:00:00Z",
+  credentialSubject: {
+    id: "did:key:z2dmzD81cgPx8Vki7JbuuMmFYrWPgYoytykUZ3eyqht1j9KbrDt4zxXoDrBWYFiATYZ8G9JMeEXC7Kki24fbTwtsJbGe5qcbkYFunSzcDokMRmj8UJ1PbdCGh33mf97K3To89bMzd15qrYq3VkDztoZqfmujkJVpvTbqoXWXqxmzNDbvMJ",
```
I don't recommend this "did key" format, I suspect its the JWK format... It might be better to just generalize this to "did:example:123"...

David-Chadwick replied:
I am using the value used by EBSI in its documentation. If I change it, the example will no longer work because the JWK referred to contains this key. I cannot change the JWK because I do not have the signing key.

OR13 commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
+  validFrom: "2021-11-01T00:00:00Z",
+  issued: "2021-10-30T00:00:00Z",
+  credentialSubject: {
+    id: "did:key:z2dmzD81cgPx8Vki7JbuuMmFYrWPgYoytykUZ3eyqht1j9KbrDt4zxXoDrBWYFiATYZ8G9JMeEXC7Kki24fbTwtsJbGe5qcbkYFunSzcDokMRmj8UJ1PbdCGh33mf97K3To89bMzd15qrYq3VkDztoZqfmujkJVpvTbqoXWXqxmzNDbvMJ",
```
```suggestion
    id: "did:example:456",
```

OR13 commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
-  }]
-  </span>
+  credentialSchema: {
+    id: "https://api-test.ebsi.eu/trusted-schemas-registry/v2/schemas/z3MgUFUkb722uq4x3dv5yAJmnNmzDFeK5UC8x83QoeLJM",
```
```suggestion
    id: "https://vendor.example/trusted-schemas-registry/v2/schemas/123",
```

OR13 commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
-  </span>
+  credentialSchema: {
+    id: "https://api-test.ebsi.eu/trusted-schemas-registry/v2/schemas/z3MgUFUkb722uq4x3dv5yAJmnNmzDFeK5UC8x83QoeLJM",
+    type: "FullJsonSchemaValidator2021",
```
can we make this `JsonSchema` per the current TR track items?

David-Chadwick replied:
Done

OR13 commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
+  },
+  expirationDate: "2031-11-30T00:00:00Z",
+  termsOfUse: {
+    id: "https://api-test.ebsi.eu/trusted-issuers-registry/v4/issuers/did:ebsi:zz7XsC9ixAXuZecoD9sZEM1/attributes/7201d95fef05f72667f5454c2192da2aa30d9e052eeddea7651b47718d6f31b0",
```
```suggestion
    id: "https://vendor.example/trusted-issuers-registry/v4/issuers/did:example:123/attributes/456",
```

David-Chadwick replied:
Not done, because the id URL actually works and can be resolved. It is a real example as asked for.

OR13 replied:
do you have permission to paint them as a target like this : )

Using a real endpoint like this could cost them money, automated crawlers will hit it, etc...

IMO its better to use W3C to host things, or to include the dereferenced content as a separate example

David-Chadwick replied:
Let's discuss this at a WG meeting. If Ivan can host the URL then we can copy the contents to there.

However, at a higher level, there is a conflict that the WG needs to resolve, between asking for a real life example and then complaining that the real life example contains a real life URL.

TallTed replied:
> [@David-Chadwick] Let's discuss this at a WG meeting

Has this been discussed and resolved? If not, I suggest creating an issue for it.

OR13 commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
-  </span>
+  credentialSchema: {
+    id: "https://api-test.ebsi.eu/trusted-schemas-registry/v2/schemas/z3MgUFUkb722uq4x3dv5yAJmnNmzDFeK5UC8x83QoeLJM",
+    type: "FullJsonSchemaValidator2021",
```
```suggestion
    type: "JsonSchema",
```

David-Chadwick replied:
Done

OR13 commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
+  id: "urn:did:123456",
+  type: ["VerifiableCredential", "VerifiableAttestation", "VerifiableId"],
+  issuer: "did:ebsi:zz7XsC9ixAXuZecoD9sZEM1",
+  issuanceDate: "2021-11-01T00:00:00Z",
```
```suggestion
  validFrom: "2021-11-01T00:00:00Z",
```

David-Chadwick replied:
Its now validFrom

OR13 commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
-      "type": "ExampleBachelorDegree",
-      "name": "Bachelor of Science and Arts"
-    }
+  id: "urn:did:123456",
```
Agree, notice the "key" side is missing quotes.  `id: "urn:did:123456",`

OR13 reviewed (changes requested):
I'm supportive of the PR, the example needs to be updated, and we will want to cover this example in tests.

to be clear, covering it in tests, just means having an input and output that looks like the example, and then after thats achieved we can remove at risk flag.

iherman commented:
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-10-04-vcwg#section1-2) on 2023-10-04

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.2. Update to Terms of Use description (pr vc-data-model#1295)

_See github pull request [vc-data-model#1295](https://github.com/w3c/vc-data-model/pull/1295)._




**David Chadwick:** no one had implemented this on the v1 data model, but there have been some now.  
… to the best of our knowledge there are several now.  
… the problem now is that the original term was generic.  
… and my original PR was to improve it to make it more encompassing.  
… the v1 had generic text, but talked about mandatory things to do...which was not equally generic.  
… so this PR makes the actions also be generic.  
… then the comments came to provide examples.  
… so I did that with real URIs, etc.  
… but then the feedback came about the URI's and that they may not be around as long as the spec.  
… so now I think the data model should have the generic text.  
… and we should point to an extension space for these.  
… so my proposal is that if we want the EBSI example to be the example, then we'd have to explain it all.  
… or we could make it a generic example...but then it wouldn't be a good example.  

**Kristina Yasuda:** so the question I have is proving that there are two implementations in the example--is that what we want to see?  
… how the use of the ToS seems underspecified.  
… so how can that be improved --that's the question I want addressed.  

**Orie Steele:** I want to thank DavidC.  
… he's been good about addressing feedback.  
… there's some v1 stuff that still needs removed, but overall this seems ready for tests to be written for.  
… once there's two tests we can merge this.  


**Manu Sporny:** the examples in the spec are non-normative, so the URL's in them shouldn't matter.  
… DavidC has proved that there are multiple implementations.  
… there is v1 cleanup.  
… but I am confused about testing the extensions.  
… feel like goal posts are being moved unless we're being very clear about what needs testing.  
… multiple people are using it.  
… there is a separate test suite.  
… and I think that tests that it's being used.  
… so not sure we need to take on the interop tests.  

**David Chadwick:** I accept that it's v1 because it's based on their existing work.  
… I can upgrade it to v2.  
… I realized it takes more than swapping out the contexts.  
… and that an additional EBSI context needs to be defined now.  
… to work with the v2 data model.  
… however, I don't think I should be describing how EBSI works.  
… maybe I can just say, this is implemented according to the EBSI spec and link there.  

**Orie Steele:** this is about testing.  
… my request is about removing that "at risk" flag.  
… to do that we need 2 supporting implementations that past tests.  
… if the only normative requirement that it use an RDF type, then having an example should pass that test.  
… I don't think there are additional tests to write.  
… just share examples that can survive issuance and validation.  

**Manu Sporny:** so Orie if I'm understanding you.  
… then a test that says, "uses issuer defined type".  
… and as long as that issues and verifies, would that meet your expectation of the testing requirement?  
… can we use an example in the type in the terms of use?  

**Orie Steele:** as far as I'm concerned we just need to prove that two implementations can handle these examples.  

**Manu Sporny:** so if 2 implementations do not choke on these examples then we're fine?  
… and you do not believe they need to understand the Terms of Use terms?  

**Orie Steele:** I hope not.  

**Manu Sporny:** k. then as long as issuance and verification don't choke across two implementations, then I agree that we're good.  

**David Chadwick:** this is great, thank you.  

**Ivan Herman:** to clarify, if DavidC goes through what was just discussed, then we remove the at-risk flag? correct?  

**Kristina Yasuda:** yes.  

**David Chadwick:** do I remove them now?  

> *Orie Steele:* I left a comment similar to what ivan said on the PR.

**Kristina Yasuda:** no, we need them tested first.  
… and an issue to track.  

</details>

David-Chadwick commented:
I have updated the example to be VCDMv2 conformant. I have also removed the inline http reference to an Informative Reference, but I cannot find the file that the references are stored in. So currently the EBSI reference is not in the Infomative References section

davidlehn commented on index.html:
```diff
@@ -3221,40 +3231,42 @@ <h3>Terms of Use</h3>
-      "type": "ExampleBachelorDegree",
-      "name": "Bachelor of Science and Arts"
+    "@context": [
+        "https://www.w3.org/ns/credentials/v2",
```
JSON should use common 2 space indent style.

David-Chadwick replied:
Done

David-Chadwick commented on index.html:
```diff
@@ -3225,36 +3235,37 @@ <h3>Terms of Use</h3>
+  issuer: "did:ebsi:zz7XsC9ixAXuZecoD9sZEM1",
+  issuanceDate: "2021-11-01T00:00:00Z",
+  validFrom: "2021-11-01T00:00:00Z",
+  issued: "2021-10-30T00:00:00Z",
```
Done

David-Chadwick commented:
> adding an example does not make ToU "better defined" and does not solve #1010
> 
The text has been improved to be more consistent in its definition and to provide examples of what it might contain. So the prohibitions etc text is now just one example of what the ToU might contain, rather than implying, as the previous text did, that this is what it should contain.

iherman commented:
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-10-11-vcwg#section1-10) on 2023-10-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.10. Update to Terms of Use description (pr vc-data-model#1295)

_See github pull request [vc-data-model#1295](https://github.com/w3c/vc-data-model/pull/1295)._




**Brent Zundel:** Raised by David Chadwick to add examples of terms of use. A number of suggested changes that folks have made, not clear if those changes have been addressed.  
… A number of reviewers requested changes. Let's talk about this PR a bit.  

**David Chadwick:** So I have made all the changes that people have requested and I have re-requested reviews.  
… There is one issue that I couldn't resolve.  
… Previously I had a reference that I changed to an informative reference, I couldn't figure out where the informative references are held ... I can actually add it properly.  
… If someone tells me what to do.  

**Manu Sporny:** There's something at the top of the document or in a file called biblio.js -- look there.  

**David Chadwick:** Ah, I didn't look there.  
… It's just JavaScript?  

**Manu Sporny:** Let me check.  
… It's in a file called "common.js".  
… If you add your link there, it will show up in the spec.  

**David Chadwick:** Ok, got it.  
… Thanks, I can sort it now.  
… I'll push that in another PR today or tomorrow.  

**Manu Sporny:** I think Kristina is concerned that this PR doesn't address -- the issue associated with this PR, there are two parts. One is to use a more accurate example, and that is done. We also wanted to demonstrate that this is broadly used and deployed and said so in the comments but Kristina's concern was that we're still not very clear when you use terms of use or how to use it.  
… I'm wondering if we can deal with that in another PR or if you can address that here, David.  

**David Chadwick:** So I have altered the text. The original text was generic, and just before the CR or after ... we put more text in around mandatory requirements. That then, throughout the whole of the text ... the text before that was put in was very generic. What I'd done is I've now created a couple of examples in generic examples to say "it may be either of the following", it doesn't pin down that terms of use must have these three prohibitions or whatever.

**David Chadwick:** It wasn't an example before and now it's an example.  
… It still has that in the second example now. The whole shift in version 1 and version 1.1 was skewed ... from the original terms of use to now go back to the original text to become more generic, so the prohibition stuff is now in an examples. We have an example where the issuer has to conform to terms of use and another example for a holder with prohibitions.  
… If Kristina can come back and say what else to change if more is needed that would be good.  

**Brent Zundel:** I see last week Kristina put herself to re-review.  
… Next step is review from Kristina and other folks, particularly those that have requested changes to see if their requests have been addressed or not.  
… Kristina, Orie, Manu are some names here.  
… Any other comments?  
… We will skip 1294 because if 1295 gets merged then 1294 will be closed.  

</details>

msporny commented on index.html:
```diff
@@ -3186,21 +3186,25 @@ <h3>Terms of Use</h3>
         <p>
-The value of the <code>termsOfUse</code> <a>property</a> tells the
-<a>verifier</a> what actions it is required to perform (an <em>obligation</em>),
+Examples for the value of the <code>termsOfUse</code> <a>property</a> may be
```
```suggestion
Examples for the value of the <code>termsOfUse</code> <a>property</a> might be
```

Avoid the use of RFC terms when not making RFC statements.

David-Chadwick replied:
done

David-Chadwick replied:
I added a different second example to replace the obligation based one that you suggested by removed. See if you like it.

msporny commented on index.html:
```diff
@@ -3186,21 +3186,25 @@ <h3>Terms of Use</h3>
-<a>verifier</a> what actions it is required to perform (an <em>obligation</em>),
+Examples for the value of the <code>termsOfUse</code> <a>property</a> may be
+to tell the <a>verifier</a> 
+<li>
```
```suggestion
<ul><li>
```

David-Chadwick replied:
done

msporny commented on index.html:
```diff
@@ -3186,21 +3186,25 @@ <h3>Terms of Use</h3>
+what procedures or policies were used in issuing the verifiable credential,
+for example, by providing a pointer to the location where these procedures 
+or policies can be found, or naming the standard that defines them.
+</li>
```
```suggestion
</li></ul>
```

David-Chadwick replied:
done

msporny commented on index.html:
```diff
@@ -3225,36 +3229,38 @@ <h3>Terms of Use</h3>
+  "id": "urn:did:123456",
+  "type": [
+    "VerifiableCredential",
+    "EbsiToUExample"
```
```suggestion
    "EbsiTermsOfUseExample"
```

Avoid too many acronyms in a row.

David-Chadwick replied:
done

msporny reviewed (approved):
Minor nits, otherwise an improvement on what's there today (replacing a fictional example with a real one).

iherman commented:
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-10-17-vcwg-special#section2-6) on 2023-10-17

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.6. Update to Terms of Use description (pr vc-data-model#1295)

_See github pull request [vc-data-model#1295](https://github.com/w3c/vc-data-model/pull/1295)._




**Brent Zundel:** This PR was raised by David Chadwick, a couple of requests for changes from Orie and Kristina. Kristina said that plainly, based on the conversation last week still has concerns but not blocking ones.  
… She is saying she won't block, Orie has a request for changes.  
… He's not on the call today, he said he's supportive on the PR, the example needs to be updated and covered in tests.  
… I believe this PR updates the example and is conformant with the data model, that part has been addressed.  
… I believe it would be appropriate to raise an issue on the test suite tracking the further concerns he has.  
… To make sure terms of use is tested appropriately.  
… That's my suggestion for this PR -- we merge it and raise an issue to track testing of `termsOfUse` in the test suite.  

> *Manu Sporny:* +1 to proposed course of action.

**Brent Zundel:** Any objections?  
… Making a note to myself, taking the action to raise a PR [issue?] for `termsOfUse` and once that's done I will indicate so, so this PR can be merged.  

</details>

brentzundel commented:
I opened https://github.com/w3c/vc-data-model-2.0-test-suite/issues/36 to track test-suite related concerns.

TallTed commented on index.html:
```diff
@@ -3225,36 +3227,38 @@ <h3>Terms of Use</h3>
-In the example above, the <a>issuer</a> (the <code>assigner</code>) is
-prohibiting <a>verifiers</a> (the <code>assignee</code>) from storing the data
-in an archive.
+In the example above, the <a>issuer</a> is asserting that it complies with the EBSI
+policies as an accredited issuer and is registered in the EBSI register of trusted issuers. 
+The <a>id</a> can be resolved by the verifier
+and this will reveal that the <a>issuer</a> has been issued an accreditation VC (in JWT format)
+by a trusted issuer higher in the EBSI trust chain [?EBSI]. 
```
```suggestion
In the example above, the <a>issuer</a> is asserting that, as a European
Blockchain Services Infrastructure (EBSI) accredited issuer, it complies with
the EBSI policies and is listed in the EBSI register of trusted issuers. The
<code>termsOfUse</code> <a>id</a> can be resolved by the verifier to check
whether the <a>issuer</a> has been issued an accreditation VC (in JWT format)
by a trusted issuer higher in the EBSI trust chain [?EBSI]. 
```

David-Chadwick replied:
done

msporny commented:
Waiting for @David-Chadwick to process @TallTed change suggestions.

brentzundel commented:
@David-Chadwick we plan to accept @TallTed 's changes and merge this PR by Friday if we don't get a response from you.

longpd reviewed (approved):
With TallTed's changes approve

iherman commented:
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-10-31-vcwg-special#section1-9) on 2023-10-31

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.9. Update to Terms of Use description (pr vc-data-model#1295)

_See github pull request [vc-data-model#1295](https://github.com/w3c/vc-data-model/pull/1295)._




> *Phillip Long:* Yay! to merging and moving on re: 1297.

**Brent Zundel:** Update to the terms of use description. Last we talked, we were in a pretty good place on this PR. Still a request for changes from Orie, some language clean up from Ted.  
… Is David on the call? I don't see him.  
… Orie's concern ... his request was regarding the test suite. I raised an issue to track those concerns. With Ted's requested changes in place, if those get added, I think this PR can be merged. That's the status as I understand it.  
… I'm wondering if it would be appropriate to give David a couple of days to accept / respond to Ted's proposed modifications and if he doesn't in a couple of days, how inappropriate would it be to just accept those modifications and merge this?  

**Manu Sporny:** I think Ted, you just requested editorial changes and I can do that. I don't think it's inappropriate to do that and it doesn't change the content.  
… I can merge all of Ted's changes if David doesn't respond in an appropriate time frame. I did ask him to respond two days ago. We can give it another two days and then I'll merge Ted's stuff and merge the PR into main as a suggested way forward.  

**Brent Zundel:** I'm going to add a comment to the issue to say we plan to accept Ted's changes and merge this PR by Friday if we don't hear from you. Is that inline with where folks are at?  

**Manu Sporny:** Yes.  

**Brent Zundel:** Ok.  

</details>

David-Chadwick commented on index.html:
```diff
@@ -3186,21 +3186,23 @@ <h3>Terms of Use</h3>
-not allowed to perform (a <em>prohibition</em>), or allowed to perform (a
-<em>permission</em>) if it is to accept the <a>verifiable credential</a> or
-<a>verifiable presentation</a>.
+Examples for the value of the <code>termsOfUse</code> <a>property</a> might be
+to tell the <a>verifier</a> 
```
I have added a real life third example from EBSI. You can find details of it here
https://api-pilot.ebsi.eu/docs/specs/trust-model/credentials-presentation-and-request
Plus I have made your grammatical changes.

David-Chadwick commented on index.html:
```diff
@@ -3186,21 +3186,23 @@ <h3>Terms of Use</h3>
+to tell the <a>verifier</a> 
+
+<ul><li>
+what procedures or policies were used in issuing the verifiable credential,
+for example, by providing a pointer to the location where these procedures 
+or policies can be found, or naming the standard that defines them;
```
done

TallTed replied:
@David-Chadwick — Wasn't done. The two words were changed, but the two punctuation changes were silently ignored. (Yes, I am now reviewing every character of my suggested changes on this PR. And wondering whether I need to do the same for other PRs. Fun!)

David-Chadwick commented on index.html:
```diff
@@ -3186,21 +3186,23 @@ <h3>Terms of Use</h3>
+or policies can be found, or naming the standard that defines them;
+</li>
+<li>
+which superior authority gave the issuer authority to issue this particular verifiable credential.
```
done

David-Chadwick commented on index.html:
```diff
@@ -3186,21 +3186,23 @@ <h3>Terms of Use</h3>
+</ul></li>
         </p>
+<p>
+The <code>termsOfUse</code> <a>property</a> SHOULD be automatically processable by the verifier.
```
Not accepted. I think we need to discuss this large addition in the WG before we accept this change

TallTed replied:
I think these few lines of editorial advisory are necessary, unless we include a complete and explicit example of how `the value of the <code>termsOfUse</code> <a>property</a>` can otherwise `be automatically processable by the verifier`, in order to satisfy the `SHOULD` in your text.

That said, I cannot create a distinct PR adding these few lines, nor a distinct issue referring to this text in the main document, until this PR is merged.

TallTed commented on index.html:
```diff
@@ -3186,21 +3186,27 @@ <h3>Terms of Use</h3>
+<p>
+The <code>termsOfUse</code> <a>property</a> SHOULD be automatically processable by the verifier.
 
         <p class="note">
```
Given that the content of this `<p class="note">` has been deleted, this `<p class="note">` and the subsequent `</p>` should also be deleted.
```suggestion
```

David-Chadwick replied:
Done. Thanks for spotting it

TallTed commented on index.html:
```diff
@@ -3186,21 +3186,27 @@ <h3>Terms of Use</h3>
-could be for the <a>subject</a> to delegate a <a>verifiable credential</a> to a
-<a>holder</a> and place terms of use restrictions on the delegated
-<a>verifiable credential</a>.
+
         </p>
```
Given that the content of this `</p>` has been deleted, the preceding `<p class="note">` and this `</p>` should also be deleted.

TallTed commented on index.html:
```diff
@@ -3186,21 +3186,27 @@ <h3>Terms of Use</h3>
-<em>permission</em>) if it is to accept the <a>verifiable credential</a> or
-<a>verifiable presentation</a>.
-        </p>
+Examples for the value of the <code>termsOfUse</code> <a>property</a> might be
+to tell the <a>verifier</a> any or all of the following:
```
```suggestion
The value of the <code>termsOfUse</code> <a>property</a> might be used to
tell the <a>verifier</a> any or all of the following, among other things:
```

David-Chadwick replied:
Done

TallTed reviewed (changes requested):
Accidentally left out earlier

David-Chadwick commented on index.html:
```diff
@@ -3186,21 +3186,27 @@ <h3>Terms of Use</h3>
+Examples for the value of the <code>termsOfUse</code> <a>property</a> might be
+to tell the <a>verifier</a> any or all of the following:
+
+<ul><li>
+the procedures or policies that were used in issuing the <a>verifiable credential</a>,
+for example, by providing a pointer to the location where these procedures 
+or policies can be found, or naming the standard that defines them;
+</li>
+<lic
+the issuer's rules and policies that govern the presentation of this 
+<a>verifiable credential</a> to the <a>verifier</a>;
+</li>
+<li>
+the identity of the entity under whose authority the issuer issued this particular <a>verifiable credential</a>.
+</ul></li>
```
done

TallTed replied:
This was *not* done. *Parts* of it were manually applied, but parts were *not*. It is incredibly frustrating that this happens repeatedly. If you use GitHub's web interface, there are single-click buttons that will add such suggestions to a batch commit, or will merge each suggestion as its own commit, and then parts of suggestions would not be lost because you didn't notice them. The HTML markup is as important as the text it formats. A `<li>` must be closed before the `<ul>` in which it occurs. `</ul></li>` is incorrect. I am concerned that I now need to review all my previously suggested changes, letter by letter, to confirm that they have been accurately applied. This does not make me happy.

TallTed commented on index.html:
```diff
@@ -3186,28 +3186,22 @@ <h3>Terms of Use</h3>
 <li>
 the identity of the entity under whose authority the issuer issued this particular <a>verifiable credential</a>.
 </ul></li>
         </p>
```
Again.
```suggestion
```

TallTed commented on index.html:
```diff
@@ -3186,28 +3186,22 @@ <h3>Terms of Use</h3>
 </li>
 <li>
 the identity of the entity under whose authority the issuer issued this particular <a>verifiable credential</a>.
 </ul></li>
```
Again.
```suggestion
</li></ul>
```

msporny commented:
Editorial, multiple reviews, changes requested and made, no objections, merging.
